android/data/src/main/java/poke/rogue/helper/data/repository/DefaultBattleRepository.kt

```diff
@@ -1,14 +1,31 @@
 package poke.rogue.helper.data.repository
 
+import android.content.Context
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+import poke.rogue.helper.data.datasource.LocalBattleDataSource
 import poke.rogue.helper.data.datasource.RemoteBattleDataSource
 import poke.rogue.helper.data.model.BattlePrediction
 import poke.rogue.helper.data.model.BattleSkill
+import poke.rogue.helper.data.model.Pokemon
+import poke.rogue.helper.data.model.PokemonWithSkill
 import poke.rogue.helper.data.model.Weather
 
-class DefaultBattleRepository(private val remoteBattleDataSource: RemoteBattleDataSource) : BattleRepository {
+class DefaultBattleRepository(
+    private val localBattleDataSource: LocalBattleDataSource,
+    private val remoteBattleDataSource: RemoteBattleDataSource,
+    private val pokemonRepository: DexRepository,
+) : BattleRepository {
+    private val cachedSkills: HashMap<Long, List<BattleSkill>> = hashMapOf()
```

Comment:
여기서 MutableMap 이 아닌 HashMap 을 사용하신 이유가 있으실까용? 

> 참고) mutableMapOf() 를 사용하면 내부적으로 LinkedHashMap()을 사용합니당

Reply:
딱히 어떤 이유는 없고 그냥 습관(?) 입니다,,,ㅎㅎㅎ
주입 순서 저장의 차이라고만 알고 있습니다만 🤔

Reply:
@JoYehyun99 반드시 적용할 내용은 아니지만 저는 `MutableMap`으로 변경하는 것을 제안드립니다.

```suggestion
    private val cachedSkills: MutableMap<Long, List<BattleSkill>> = hashMapOf()
```

반드시 적용할 내용은 아니지만 `MutableMap` 바꿔주셨으면 좋겠습니다!  

이유 1) 구현체 타입보다는 인터페이스 타입인 `MutableMap`으로 추상화하는 것이 추후 ConcurrentHashMap 이나 다른 LinkedHashMap 으로 바뀔 경우 더 유연하게 바꿀 수 있을 것입니다!  

---

참고) LinkedHashMap 가 필요한 경우가 뭐가 있을까 생각해보았는데, 아마 이를 활용해서 [LRU 캐시](https://velog.io/@haero_kim/LRU-Cache-%EC%9D%B4%ED%95%B4%ED%95%98%EA%B8%B0)를 구현할 때 사용할 수 있을 것 같습니다.(쉽게 말해 가장 오래 사용안한 데이터 부터 삭제하는 캐시 전략입니다)

LinkedHashMap 은 `accessOrder` 라는 옵션과 사이즈 크기의 limit 를 지정해줄 수 있어 LRU 캐시를 구현할 수 있습니다.